cse/AuthorMappingWriter.py

```diff
@@ -0,0 +1,65 @@
+import csv
+import os
+
+class AuthorMappingWriter(object):
+
+    __delimiter = ''
+    __filepath = ""
+    __file = None
+    __writer = None
+
+    def __init__(self, filepath, delimiter=','):
+        self.__delimiter = delimiter
+        self.__filepath = filepath
+
+
+    def open(self):
+        if not os.path.exists(os.path.dirname(self.__filepath)):
+            try:
+                os.makedirs(os.path.dirname(self.__filepath))
+            except OSError as exc: # Guard against race condition
+                if exc.errno != errno.EEXIST:
+                    raise
+
+        # w  = writing, will empty file and write from beginning (file is created)
+        # a+ = read and append (file is created if it does not exist)
+        self.__file = open(self.__filepath, 'w', newline='')   
+        self.__writer = csv.writer(self.__file)
+        return self
+
+
+    def close(self):
+        self.__file.close()
+
+
+    def printHeader(self, template=None):
+        if template is None:
+            self.__writer.writerow(["author_id", "author"])
+        else:
+            self.__writer.writerow(template)
+        self.__file.flush()
+
+
+    def printData(self, data):
+        for author in data:
+            self.__writer.writerow([
+                str(data[author]),
+                author,
```

Comment:
why do you use a list in `ArticleIdWriter` and a dict in `AuthorMappingIdWriter`, shouldn't it be more consistent? I guess, it's the same kind of task (creating a new mapping: string -> integer) for all three mappings

Reply:
You need to check if a user already has an id. The lookup with dict is much faster (O(1)) than  within a list (O(n))

Reply:
That's clear to me, but this does not affect the way the mappings are written to disk, is it?

Reply:
Of course it changes the way you have to write the data.... Or I don't see your point

Reply:
Resolved